bottlecap/src/main.rs

```diff
@@ -96,17 +102,23 @@ fn register() -> Result<RegisterResponse> {
 
     let resp = ureq::post(&url)
         .set(EXTENSION_NAME_HEADER, EXTENSION_NAME)
+        .set(EXTENSION_ACCEPT_FEATURE_HEADER, EXTENSION_FEATURES)
         .send_json(ureq::json!(map))
         .map_err(|e| Error::new(std::io::ErrorKind::InvalidData, e.to_string()))?;
 
     if resp.status() != 200 {
         panic!("Unable to register extension")
     }
 
-    let ext_id = resp.header(EXTENSION_ID_HEADER).unwrap_or_default();
-    Ok(RegisterResponse {
-        extension_id: ext_id.to_string(),
-    })
+    let extension_id = resp
+        .header(EXTENSION_ID_HEADER)
+        .unwrap_or_default()
+        .to_string();
+    let mut register_response = resp.into_json::<RegisterResponse>()?;
+
+    register_response.extension_id = extension_id;
```

Comment:
Why the skip_serializing on the extension_id field, and why do we overwrite it here?

Reply:
It comes as a header, not available in the payload.